core/primitives/src/crypto.rs

```diff
@@ -266,17 +268,17 @@ impl<T: AsMut<[u8]> + AsRef<[u8]> + Default + Derive> Ss58Codec for T {
 ///
 /// For now it just specifies how to create a key from a phrase and derivation path.
 #[cfg(feature = "std")]
-pub trait Pair: Sized {
+pub trait Pair: Sized + Send + Sync + 'static {
```

Comment:
likewise here -- i don't see any reason why these bounds have to exist on `Pair`.

Reply:
Fixed.